### PR TITLE
fix : File with extensions for other files should not be set to hlasm

### PR DESCRIPTION
**Pull request by SWETAS04**

### CODE REVIEW

File: clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts [hunk 1]
```diff
@@ -38,14 +38,22 @@ export class HLASMLanguageDetection {
...
                 vscode.languages.setTextDocumentLanguage(document, 'hlasm');
                 return true;
             }
         }
         return document.languageId == 'hlasm';
     }
+    //Checks for extension for plaintext files
+    EndsWithExtension(document: vscode.TextDocument): boolean {
+
+        var regexExtension = new RegExp("\\.\\w+$");
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts:52]: maybe https://nodejs.org/api/path.html#path_path_extname_path ?

Comment (SWETAS04) [reply clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts:52]: Cobol Explorer has naming conventions like dataset(member). extension .
Ex :ZIMLU02.NEW.COBOL(PARTSUPP).cbl. This api will return true even for files without extension.

Comment (slavek-kucera) [reply clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts:52]: ok, got it (I think), but we should probably still handle cases like '.hidden' as not having extension.

File: clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts [hunk 2]
```diff
@@ -38,14 +38,22 @@ export class HLASMLanguageDetection {
...
                 vscode.languages.setTextDocumentLanguage(document, 'hlasm');
                 return true;
             }
         }
         return document.languageId == 'hlasm';
     }
+    //Checks for extension for plaintext files
+    EndsWithExtension(document: vscode.TextDocument): boolean {
+
+        var regexExtension = new RegExp("^[^\\.].*\\.\\w+$");
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts:52]:
```suggestion
        var regexExtension = new RegExp("[^.].*\\.\\w+$");
```
`.hidden` => no ext
`.hidden.cbl` = > `.cbl`?

Comment (SWETAS04) [reply clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts:52]: Yes, right .

File: clients/vscode-hlasmplugin/src/test/suite/hlasmLanguageDetection.test.ts [hunk 3]
```diff
@@ -39,16 +39,96 @@ suite('Language Detection Test Suite', () => {
...
+		document.uri = vscode.Uri.file('ZIMLU02.FTP.COBOL(BIGFILE)');
+		assert.ok(!detector.EndsWithExtension(document));
+	});
+	test('Cobol file ending with extension', async () => {
+		var document = new TextDocumentMock();
+		// set plain text file to HLASM
+		document.languageId = 'plaintext';
+		document.text = nonHlasmContents;
+		document.fileName = "ZIMLU02.FTP.COBOL(BIGFILE).cbl";
+		document.uri = vscode.Uri.file('ZIMLU02.FTP.COBOL(BIGFILE).cbl');
```
Comment (michalbali256) [inline clients/vscode-hlasmplugin/src/test/suite/hlasmLanguageDetection.test.ts:85]:
```suggestion
		document.fileName = "USER.FTP.COBOL(BIGFILE)";
		document.uri = vscode.Uri.file('USER.FTP.COBOL(BIGFILE)');
		assert.ok(!detector.EndsWithExtension(document));
	});
	test('Cobol file ending with extension', async () => {
		var document = new TextDocumentMock();
		// set plain text file to HLASM
		document.languageId = 'plaintext';
		document.text = nonHlasmContents;
		document.fileName = "USER.FTP.COBOL(BIGFILE).cbl";
		document.uri = vscode.Uri.file('USER.FTP.COBOL(BIGFILE).cbl');
```

File: clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts [hunk 4]
```diff
@@ -38,14 +38,22 @@ export class HLASMLanguageDetection {
...
         }
         return document.languageId == 'hlasm';
     }
+    //Checks for extension for plaintext files
+    EndsWithExtension(document: vscode.TextDocument): boolean { 
+        
+        var regexExtension = new RegExp("[^.].*\\.\\w+$");
+        if (regexExtension.test(path.basename(document.fileName)))
+            return true;
+        return false;
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts:55]:
```suggestion
        return /^\.\w+$/.test(path.extname(document.fileName));
```

**Review by slavek-kucera — APPROVED** [review]
Please add a note to clients/vscode-hlasmplugin/CHANGELOG.md.

**Review by michalbali256 — APPROVED** [review]

**Pull request merged**